File: App/app/src/main/java/com/github/travelplannerapp/travels/TravelsModule.java
package com.github.travelplannerapp.travels;

import dagger.Module;
import dagger.Provides;

/**
 * Define LoginActivity-specific dependencies here.
 */
@Module
public class TravelsModule {

    @Provides
    TravelsPresenter provideLobbyPresenter(TravelsContract.View travelsView) {
        return new TravelsPresenter(travelsView);
    }
}

File: App/app/src/main/java/com/github/travelplannerapp/travels/TravelsModule.kt
package com.github.travelplannerapp.travels

import dagger.Module
import dagger.Provides

/**
 * Define TravelsActivity-specific dependencies here.
 */
@Module
class TravelsModule {

    @Provides
    internal fun provideTravelsPresenter(travelsView: TravelsContract.View): TravelsContract.Presenter {
        return TravelsPresenter(travelsView)
    }
}